### PR TITLE
remove flags

### PR DESCRIPTION
**Pull request by ruancomelli**
## Summary by Sourcery

This pull request refactors the code by removing the use of bitwise flag operations and replacing them with individual boolean member variables. This change improves code readability and maintainability. Corresponding updates have been made to the unit tests to reflect these changes.

- **Enhancements**:
    - Replaced the use of bitwise flag operations with individual boolean member variables for better readability and maintainability.
- **Tests**:
    - Updated unit tests to reflect the removal of flag operations and the introduction of boolean member variables.

### CODE REVIEW

**Comment by ruancomelli** [comment]
This is a benchmark review for experiment `improve_review_guide`.
Run ID: `improve_review_guide/benchmark_2024-08-06T17-38-05_v1-21-0-70-g56484f776-dirty`.

This pull request was cloned from `https://github.com/sourcery-ai-experiments/iptux/pull/6`. _(Note: the URL is not a link to avoid triggering a notification on the original pull request.)_
<details>
<summary>Experiment configuration</summary>

```yaml
review_config:
  # User configuration for the review
  # - benchmark - use the user config from the benchmark reviews
  # - <value> - use the value directly
  user_review_config:
    enable_ai_review: true
    enable_rule_comments: false

    enable_complexity_comments: false
    enable_security_comments: false
    enable_tests_comments: false
    enable_comment_suggestions: false
    enable_functionality_review: false

    enable_pull_request_summary: false
    enable_review_guide: true

    enable_approvals: false

  ai_review_config:
    # The model responses to use for the experiment
    # - benchmark - use the model responses from the benchmark reviews
    # - llm - call the language model to generate responses
    model_responses:
      comments_model: benchmark
      comment_area_model: benchmark
      comment_validation_model: benchmark
      comment_suggestion_model: benchmark
      complexity_model: benchmark
      docstrings_model: benchmark
      functionality_model: benchmark
      security_model: benchmark
      tests_model: benchmark
      pull_request_summary_model: benchmark
      review_guide_model: llm

# The pull request dataset to run the experiment on
pull_request_dataset:
  # open-source PRs
- https://github.com/sourcery-ai-experiments/gpt-crawler/pull/1
- https://github.com/sourcery-ai-experiments/graphviz2drawio/pull/1
- https://github.com/sourcery-ai-experiments/graphviz2drawio/pull/2
- https://github.com/sourcery-ai-experiments/iptux/pull/5
- https://github.com/sourcery-ai-experiments/iptux/pull/6
- https://github.com/sourcery-ai-experiments/kalima/pull/3
- https://github.com/sourcery-ai-experiments/PED/pull/3
- https://github.com/sourcery-ai-experiments/pydantic_tfl_api/pull/1
- https://github.com/sourcery-ai-experiments/pydantic_tfl_api/pull/2
- https://github.com/sourcery-ai-experiments/Trenova/pull/11
  # Sourcery PRs
- https://github.com/sourcery-ai/core/pull/4055
- https://github.com/sourcery-ai/core/pull/4263
- https://github.com/sourcery-ai/core/pull/4352
- https://github.com/sourcery-ai/core/pull/4364
- https://github.com/sourcery-ai/core/pull/4366
- https://github.com/sourcery-ai/core/pull/4390
- https://github.com/sourcery-ai/core/pull/4402
- https://github.com/sourcery-ai/core/pull/4445
- https://github.com/sourcery-ai/core/pull/4462
- https://github.com/sourcery-ai/core/pull/4463

review_comment_labels: []

# Benchmark reviews generated by running
#   python -m scripts.experiment benchmark <experiment_name>
benchmark_reviews: []

```

</details>

**Comment by SourceryAI** [comment]
## Reviewer's Guide by Sourcery

This pull request refactors the code by removing the use of bitwise flag operations and replacing them with individual boolean member variables. This change improves code readability and maintainability. Corresponding updates have been made to the unit tests to reflect these changes.

### File-Level Changes

| Files | Changes |
| ----- | ------- |
| `src/iptux-core/ProgramData.cpp`<br/>`src/iptux-core/Models.cpp`<br/>`src/api/iptux-core/ProgramData.h`<br/>`src/api/iptux-core/Models.h` | Replaced bitwise flag operations with individual boolean member variables for better readability and maintainability. |
| `src/iptux-utils/utils.cpp`<br/>`src/iptux-utils/UtilsTest.cpp`<br/>`src/iptux-utils/utils.h` | Removed FLAG_SET, FLAG_CLR functions, and related tests. |
| `src/iptux/DataSettings.cpp`<br/>`src/iptux/MainWindow.cpp` | Updated method calls to use new boolean member variables. |

---

<details>
<summary>Tips</summary>

- Trigger a new Sourcery review by commenting `@sourcery-ai review` on the pull request.
- Continue your discussion with Sourcery by replying directly to review comments.
- You can change your review settings at any time by accessing your [dashboard](https://app.sourcery.ai):
    - Enable or disable the Sourcery-generated pull request summary or reviewer's guide;
    - Change the review language;
- You can always [contact us](mailto:support@sourcery.ai) if you have any questions or feedback.

</details>

File: src/api/iptux-core/Models.h [hunk 1]
```diff
@@ -134,7 +134,7 @@ class PalInfo {
   PalInfo& setCompatible(bool value);
   PalInfo& setOnline(bool value);
   PalInfo& setChanged(bool value);
-  PalInfo& setInBlacklistl(bool value);
```
Comment (SourceryAI) [inline src/api/iptux-core/Models.h:137]: **issue (typo):** Fix typo in method name setInBlacklistl.

Rename the method setInBlacklistl to setInBlacklist to match the naming convention of the other methods.

**Review by SourceryAI — COMMENTED** [review]
Hey @ruancomelli - I've reviewed your changes - here's some feedback:

**Overall Comments**:
- Consider using `bool` directly for the new member variables instead of `uint8_t` with bit fields. This will make the code more idiomatic and easier to understand.
- Ensure that all instances of the removed `FLAG_SET` and `FLAG_ISSET` macros are properly replaced and tested to avoid any potential issues.

<details>
<summary>Here's what I looked at during the review</summary>

- 🟡 **General issues**: 1 issue found
- 🟢 **Documentation**: all looks good
</details>

***

[LangSmith trace](https://smith.langchain.com/o/7d8518ad-6dd3-5c7b-96d2-757769296335/projects/p/88560225-5072-48bc-967c-dc54da53f265/r/6291dbb9-9d5d-4041-bd47-ac7e23db30ba?poll=true)

<sub>
Help me be more useful! Please click 👍 or 👎 on each comment to tell me if it was helpful.
</sub>

**Pull request closed without merging**